core/src/main/scala/codesearch/core/regex/lexer/Tokenizer.scala

```diff
@@ -10,15 +10,17 @@ object Tokenizer {
 
   private def endForCharSet[_: P] = P("]").!
 
+  private def charSetPred[_: P] = P("[:" ~ (!":]" ~ AnyChar).rep ~ ":]")
+
   private def specialSymbols[_: P] =
     P("\\" | " " | "." | "|" | "$" | "%" | "^" | "&" | "*" | "+" | "?" | "!" | "[" | "]" | "{" | "}" | "(" | ")").!
 
   private def parserEscaped[_: P] = P("\\" ~ AnyChar.!).map(a => Escaped(a.charAt(0)))
 
-  private def parserCharInsideSet[_: P] = P(!"\\" ~ !endForCharSet ~ AnyChar).rep.!
+  private def parserCharInsideSet[_: P] = P(!"\\" ~ !endForCharSet ~ (charSetPred | AnyChar)).rep.!
```

Comment:
This is incorrect; escaped characters can also be in character sets. E.g. `[\n]` is valid.

Please fix it _and_ add a test-case for it.